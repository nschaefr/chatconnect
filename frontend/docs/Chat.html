<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>Chat - Documentation</title>

    <script src="scripts/prettify/prettify.js"></script>
    <script src="scripts/prettify/lang-css.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link
      type="text/css"
      rel="stylesheet"
      href="https://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css"
    />
    <link
      type="text/css"
      rel="stylesheet"
      href="styles/prettify-tomorrow.css"
    />
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css" />
  </head>
  <body>
    <input type="checkbox" id="nav-trigger" class="nav-trigger" />
    <label for="nav-trigger" class="navicon-button x">
      <div class="navicon"></div>
    </label>

    <label for="nav-trigger" class="overlay"></label>

    <nav>
      <li class="nav-link nav-home-link"><a href="index.html">Home</a></li>
      <li class="nav-heading">Classes</li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"><a href="Avatar.html">Avatar</a></span>
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Avatar.html#~getColorByHash">getColorByHash</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Avatar.html#~hashCode">hashCode</a></span
        >
      </li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"><a href="Chat.html">Chat</a></span>
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~connectToWebSocket">connectToWebSocket</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~filterbyKeyword">filterbyKeyword</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~handleMessage">handleMessage</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~handleResize">handleResize</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~logout">logout</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~sendMessage">sendMessage</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~showOnlinePeople">showOnlinePeople</a></span
        >
      </li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"><a href="Contact.html">Contact</a></span>
      </li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"><a href="Routes.html">Routes</a></span>
      </li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"
          ><a href="SignuploginForm.html">SignuploginForm</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="SignuploginForm.html#~submitHandler">submitHandler</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="SignuploginForm.html#~validatePassword"
            >validatePassword</a
          ></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="SignuploginForm.html#~validateUsername"
            >validateUsername</a
          ></span
        >
      </li>
    </nav>

    <div id="main">
      <h1 class="page-title">Chat</h1>

      <section>
        <header>
          <h2>Chat</h2>
        </header>

        <article>
          <div class="container-overview">
            <div class="section-method">
              <h4 class="name" id="Chat">
                <span class="type-signature"></span>new Chat<span
                  class="signature"
                  >()</span
                ><span class="type-signature"></span>
              </h4>

              <div class="description">
                <p>The Page where the chatting happens</p>
              </div>

              <dl class="details">
                <dt class="tag-source">Source:</dt>
                <dd class="tag-source">
                  <ul class="dummy">
                    <li>
                      <a href="components_chat-page_chat.jsx.html"
                        >components/chat-page/chat.jsx</a
                      >,
                      <a href="components_chat-page_chat.jsx.html#line18"
                        >line 18</a
                      >
                    </li>
                  </ul>
                </dd>
              </dl>
            </div>
          </div>

          <h3 class="subsection-title">Methods</h3>

          <div class="section-method">
            <h4 class="name" id="~connectToWebSocket">
              <span class="type-signature">(inner) </span
              >connectToWebSocket<span class="signature">()</span
              ><span class="type-signature"></span>
            </h4>

            <div class="description">
              <p>Connects the user to the websocket</p>
            </div>

            <dl class="details">
              <dt class="tag-source">Source:</dt>
              <dd class="tag-source">
                <ul class="dummy">
                  <li>
                    <a href="components_chat-page_chat.jsx.html"
                      >components/chat-page/chat.jsx</a
                    >,
                    <a href="components_chat-page_chat.jsx.html#line46"
                      >line 46</a
                    >
                  </li>
                </ul>
              </dd>
            </dl>
          </div>

          <div class="section-method">
            <h4 class="name" id="~filterbyKeyword">
              <span class="type-signature">(inner) </span>filterbyKeyword<span
                class="signature"
                >()</span
              ><span class="type-signature"></span>
            </h4>

            <div class="description">
              <p>
                If a user searchs for an other user it filters the users by the
                keyword typed in by the user
              </p>
            </div>

            <dl class="details">
              <dt class="tag-source">Source:</dt>
              <dd class="tag-source">
                <ul class="dummy">
                  <li>
                    <a href="components_chat-page_chat.jsx.html"
                      >components/chat-page/chat.jsx</a
                    >,
                    <a href="components_chat-page_chat.jsx.html#line143"
                      >line 143</a
                    >
                  </li>
                </ul>
              </dd>
            </dl>
          </div>

          <div class="section-method">
            <h4 class="name" id="~handleMessage">
              <span class="type-signature">(inner) </span>handleMessage<span
                class="signature"
                >(event)</span
              ><span class="type-signature"></span>
            </h4>

            <div class="description">
              <p>Handles messages</p>
            </div>

            <dl class="details">
              <dt class="tag-source">Source:</dt>
              <dd class="tag-source">
                <ul class="dummy">
                  <li>
                    <a href="components_chat-page_chat.jsx.html"
                      >components/chat-page/chat.jsx</a
                    >,
                    <a href="components_chat-page_chat.jsx.html#line82"
                      >line 82</a
                    >
                  </li>
                </ul>
              </dd>
            </dl>

            <h5>Parameters:</h5>

            <table class="params">
              <thead>
                <tr>
                  <th>Name</th>

                  <th>Type</th>

                  <th class="last">Description</th>
                </tr>
              </thead>

              <tbody>
                <tr>
                  <td class="name"><code>event</code></td>

                  <td class="type">
                    <span class="param-type"><code>string</code></span>
                  </td>

                  <td class="description last"></td>
                </tr>
              </tbody>
            </table>
          </div>

          <div class="section-method">
            <h4 class="name" id="~handleResize">
              <span class="type-signature">(inner) </span>handleResize<span
                class="signature"
                >()</span
              ><span class="type-signature"></span>
            </h4>

            <div class="description">
              <p>
                A Function which decides if the App shows in mobile view or
                normal browser view
              </p>
            </div>

            <dl class="details">
              <dt class="tag-source">Source:</dt>
              <dd class="tag-source">
                <ul class="dummy">
                  <li>
                    <a href="components_chat-page_chat.jsx.html"
                      >components/chat-page/chat.jsx</a
                    >,
                    <a href="components_chat-page_chat.jsx.html#line36"
                      >line 36</a
                    >
                  </li>
                </ul>
              </dd>
            </dl>
          </div>

          <div class="section-method">
            <h4 class="name" id="~logout">
              <span class="type-signature">(inner) </span>logout<span
                class="signature"
                >()</span
              ><span class="type-signature"></span>
            </h4>

            <div class="description">
              <p>Logs out the user</p>
            </div>

            <dl class="details">
              <dt class="tag-source">Source:</dt>
              <dd class="tag-source">
                <ul class="dummy">
                  <li>
                    <a href="components_chat-page_chat.jsx.html"
                      >components/chat-page/chat.jsx</a
                    >,
                    <a href="components_chat-page_chat.jsx.html#line132"
                      >line 132</a
                    >
                  </li>
                </ul>
              </dd>
            </dl>
          </div>

          <div class="section-method">
            <h4 class="name" id="~sendMessage">
              <span class="type-signature">(inner) </span>sendMessage<span
                class="signature"
                >(event)</span
              ><span class="type-signature"></span>
            </h4>

            <div class="description">
              <p>Sends messages typed in and submittet by the user</p>
            </div>

            <dl class="details">
              <dt class="tag-source">Source:</dt>
              <dd class="tag-source">
                <ul class="dummy">
                  <li>
                    <a href="components_chat-page_chat.jsx.html"
                      >components/chat-page/chat.jsx</a
                    >,
                    <a href="components_chat-page_chat.jsx.html#line96"
                      >line 96</a
                    >
                  </li>
                </ul>
              </dd>
            </dl>

            <h5>Parameters:</h5>

            <table class="params">
              <thead>
                <tr>
                  <th>Name</th>

                  <th>Type</th>

                  <th class="last">Description</th>
                </tr>
              </thead>

              <tbody>
                <tr>
                  <td class="name"><code>event</code></td>

                  <td class="type">
                    <span class="param-type"><code>*</code></span>
                  </td>

                  <td class="description last"></td>
                </tr>
              </tbody>
            </table>
          </div>

          <div class="section-method">
            <h4 class="name" id="~showOnlinePeople">
              <span class="type-signature">(inner) </span>showOnlinePeople<span
                class="signature"
                >(peopleArray)</span
              ><span class="type-signature"></span>
            </h4>

            <div class="description">
              <p>Shows the user which other users are online</p>
            </div>

            <dl class="details">
              <dt class="tag-source">Source:</dt>
              <dd class="tag-source">
                <ul class="dummy">
                  <li>
                    <a href="components_chat-page_chat.jsx.html"
                      >components/chat-page/chat.jsx</a
                    >,
                    <a href="components_chat-page_chat.jsx.html#line60"
                      >line 60</a
                    >
                  </li>
                </ul>
              </dd>
            </dl>

            <h5>Parameters:</h5>

            <table class="params">
              <thead>
                <tr>
                  <th>Name</th>

                  <th>Type</th>

                  <th class="last">Description</th>
                </tr>
              </thead>

              <tbody>
                <tr>
                  <td class="name"><code>peopleArray</code></td>

                  <td class="type">
                    <span class="param-type"
                      ><code>Array.&lt;string></code></span
                    >
                  </td>

                  <td class="description last">
                    <p>online users</p>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </article>
      </section>
    </div>

    <br class="clear" />

    <footer>
      Generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 4.0.2</a> on
      Thu Sep 07 2023 18:27:20 GMT+0200 (Central European Summer Time) using the
      Minami theme.
    </footer>

    <script>
      prettyPrint();
    </script>
    <script src="scripts/linenumber.js"></script>
  </body>
</html>
