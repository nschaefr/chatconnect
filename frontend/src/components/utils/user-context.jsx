import { createContext, useState } from "react";

export const UserContext = createContext({});

export function ContextProvider({ children }) {
  const [username, setLoggedInUsername] = useState();
  const [id, setId] = useState();

  return (
    <UserContext.Provider value={{ username, setLoggedInUsername, id, setId }}>
      {children}
    </UserContext.Provider>
  );
}
