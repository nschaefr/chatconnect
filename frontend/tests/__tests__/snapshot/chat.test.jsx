import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Chat from "../../../src/components/chat-page/chat";
import { UserContext } from "../../../src/components/utils/user-context";

const UserContextMock = ({ children }) => (
  <UserContext.Provider
    value={{
      username: "mockuser",
      setLoggedInUsername: jest.fn(),
      id: "123",
      setId: jest.fn(),
    }}
  >
    {children}
  </UserContext.Provider>
);

describe("chat snapshot-tests", () => {
  it("rendering chat", () => {
    const { container } = render(
      <UserContextMock>
        <Chat />
      </UserContextMock>
    );

    expect(container).toMatchSnapshot();
  });
});
