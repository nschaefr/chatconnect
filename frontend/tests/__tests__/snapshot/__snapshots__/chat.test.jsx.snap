// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat snapshot-tests rendering chat 1`] = `
<div>
  <div
    class="chatDiv"
    data-testid="chat-component"
  >
    <div
      class="sidebarDiv"
    >
      <div
        class="informationDiv"
      >
        <div
          class="logoDiv"
        >
          <img
            src="[object Object]"
            width="35px"
          />
          <p
            class="appName"
          >
            chatconnect
          </p>
        </div>
        <div
          class="searchBarDiv"
        >
          <input
            class="searchBar"
            placeholder="Find contacts..."
          />
        </div>
        <div />
      </div>
      <div
        class="bottomDiv"
      >
        <div
          class="userInformation"
        >
          <div
            style="background-color: rgb(115, 182, 165); width: 30px; height: 30px; display: flex; border-radius: 50%; align-items: center; justify-content: center;"
          >
            <div
              style="font-family: ExtraBold; color: white; font-size: 13px;"
            >
              M
            </div>
          </div>
          <div
            class="usernameInformationDiv"
          >
            <div
              class="usernameInformation"
            >
              <span
                class="username"
              >
                mockuser
              </span>
            </div>
          </div>
        </div>
        <div
          class="logoutDiv"
          data-testid="logout-button"
        >
          <img
            src="[object Object]"
            width="30px"
          />
        </div>
      </div>
    </div>
    <div
      class="chattingDiv"
    >
      <div
        class="contentAreaDiv"
      >
        <div
          class="noSelectedContact"
        >
          Select a chat on the left.
        </div>
      </div>
    </div>
  </div>
</div>
`;
